local-cluster/tests/local_cluster.rs

```diff
@@ -4257,7 +4262,10 @@ fn test_leader_failure_4() {
     solana_logger::setup_with_default(RUST_LOG_FILTER);
     error!("test_leader_failure_4");
     let num_nodes = 4;
-    let validator_config = ValidatorConfig::default_for_test();
+    let validator_config = ValidatorConfig {
+        block_production_method: BlockProductionMethod::ThreadLocalMultiIterator,
```

Comment:
This test doesn't "rely" on forwarding, but is slowed down without it.

There's a loop in here which loops over all nodes, creates a client with that node and will send txs to that node until confirmed.
So when that node is not leader, there is no progress, since it doesn't get forwarded the leader.